### DISCUSSION TITLE
After an update of the ltsp package, what to do and in what order

### DISCUSSION THREAD

**Discussion by rkwesk** [ltsp/ltsp#181] [Q&A]
Updates of the ltsp package are understandably frequent at the moment but imho this issue will remain relevant even later on (and would probably warrant additional documentation now.)

The issue:

After receiving the new ltsp package we know we need to

   ltsp image /

and

   ltsp initrd

But sometimes we also need

   ltsp -o ipxe

and depending on something new we may wish to edit /etc/ltsp/ltsp.conf and/or /etc/dnsmasq.d/ltsp-dnsmasq.conf

or

   ltsp -o dnsmasq (followed by whatever parameters)

In short I'm asking for advice for us all insofar as what to do and it what order to do it in.

Richard

**Comment by alkisg** [comment]
`ltsp image /` is rarely needed on ltsp updates, because the code update is also sent via the initramfs (`ltsp initrd`).

The following may or may not be needed, depending on the update:

```shell
apt full-upgrade
ltsp initrd
ltsp -o ipxe
ltsp -o dnsmasq
ltsp -o nfs
```

On each bug report / solution, I mention which of those are needed. And of course one could just run all of them and avoid reading bug reports / changelogs! :smile:

**Comment by rkwesk** [comment]
Thank you for clearing that up. What do you think about adding this to the documentation?

Richard

**Comment by alkisg** [comment]
Nah, I think it would be confusing to add it now and then remove it a couple of months later when things are more stable.

**Comment by dgroos** [comment]
> ltsp -o dnsmasq

Assuming one doesn't know for sure if it's needed, should one run the following command on both a 2 NIC and 1 NIC system after an update?

**Comment by alkisg** [comment]
I'm thinking it would be easier for the simple use cases to assume that the sysadmin doesn't customize those files. Which means we can default to OVERWRITE=1 and omit the -o everywhere. So no, running all these commands shouldn't cause any issues even when they're not needed.
